pallets/crowdloans/src/lib.rs

```diff
@@ -959,6 +979,122 @@ pub mod pallet {
             }
             Ok(().into())
         }
+
+        #[pallet::weight(<T as Config>::WeightInfo::refund())]
+        #[transactional]
+        pub fn refund(origin: OriginFor<T>, crowdloan: ParaId) -> DispatchResult {
```

Comment:
Since failed crowdloan can open a new one. So maybe the signature should be:

```rust
refund(origin: OriginFor<T>, crowdloan: ParaId, leaseStart: LeasePeriod, leaseEnd: LeasePeriod)
```

Reply:
and you can use `Self::vaults((&crowdloan, &leaseStart, &leaseEnd))` to get vault instead of `current_vault

Reply:
Would like to know the test case for this ?